fractal_tasks_core/lib_metadata_parsing.py

```diff
@@ -85,23 +98,44 @@ def parse_yokogawa_metadata(
             f"There were {error_count} ERR entries in the metadatafile. "
             f"Still succesfully parsed {len(site_metadata)} sites. "
         )
-    total_files = len(mlf_frame)
-    # TODO: Check whether the total_files correspond to the number of
-    # relevant input images in the input folder. Returning it for now
-    # Maybe return it here for further checks and produce a warning if it does
-    # not match
 
-    return site_metadata, total_files
+    # Compute expected number of image files for each well
+    list_of_wells = set(site_metadata.index.get_level_values("well_id"))
+    number_of_files = {}
+    for well_id in list_of_wells:
+        num_images = mlf_frame.MeasurementRecord.str.contains(
+            f"_{well_id}_"
```

Comment:
Yes, as you already raised @tcompa , this may not be restrictive enough. We could e.g. have an image file called: `Test_Acquisition_Plate_B03_experimental_C02_T0001F0001L01A01Z01C01.tif`
In that case, the expected number of files for well B03 would also count this file => the file count check would fail
We can be more strict by only looking for well entries before the TFLAZC string that always follows the filename. I ran this on a few examples, this seems to do the trick:
`f"_{well_id}_T.*F.*L.*A.*Z.*C"`

Reply:
Or if we want to be even more strict, we should use '_C02_T[0-9]*F[0-9]*L[0-9]*A[0-9]*Z[0-9]*C[0-9]*' (to enforce that "Test_Acquisition_Plate_B03_T_experimental_C02_T0001F0001L01A01Z01C01.tif" is not a match for well B03)

Reply:
I'd rather skip additional complex parsing, and in fact we should probably just go with the straightforward
```python
    for this_well_id in list_of_wells:
        num_images = (mlf_frame.well_id == this_well_id).sum()
```
which looks much more reliable.

Anyway, I'll write a multi-well test, just to be sure that this is under control

Reply:
Ah, even better :)